webpack/webpack-deep-imports-compat-plugin/src/DeepImportsCompatPlugin.ts

```diff
@@ -0,0 +1,271 @@
+      this._virtualModules.writeModule(this._moduleName, lines);
+    });
+
+    compiler.hooks.thisCompilation.tap(PLUGIN_NAME, (compilation) => {
+      compilation.hooks.processAssets.tapPromise(PLUGIN_NAME, async () => {
+        // `ChunkGroup` is not exported from 'webpack'
+        // eslint-disable-next-line @typescript-eslint/typedef
+        const chunkGroup /* ChunkGroup | undefined */ = compilation.namedChunkGroups.get(bundleName);
+        if (!chunkGroup) {
+          throw new Error(`The chunk group ${bundleName} was not found.`);
+        }
+
+        let runtimeChunk: Chunk | undefined;
+        for (const chunk of chunkGroup.chunks) {
+          if (chunk.hasRuntime()) {
+            if (runtimeChunk) {
+              throw new Error(`Multiple runtime chunks were found in the ${bundleName} chunk group.`);
+            } else {
+              runtimeChunk = chunk;
+            }
+          }
+        }
+
+        if (!runtimeChunk) {
+          throw new Error(`The runtime chunk in the ${bundleName} chunk group not found.`);
+        }
+
+        const filenames: string[] = Array.from(runtimeChunk.files);
+        let jsFileBaseName: string | undefined;
+        for (const filename of filenames) {
+          if (filename.endsWith(JS_EXTENSION)) {
+            if (jsFileBaseName) {
+              throw new Error(`Multiple JS files were found in the ${bundleName} chunk group.`);
+            } else {
+              jsFileBaseName = filename.substring(0, filename.length - JS_EXTENSION.length);
+            }
+          }
+        }
+
+        if (!jsFileBaseName) {
+          throw new Error(`The JS file in the ${bundleName} chunk group not found.`);
+        }
+
+        const outputPath: string | undefined = compilation.options.output.path;
+        if (!outputPath) {
+          throw new Error(`The "output.path" option was not specified.`);
+        }
+
+        const jsFilePath: string = Path.convertToSlashes(path.join(outputPath, jsFileBaseName));
+        const jsFileFolderPath: string = jsFilePath.substring(0, jsFilePath.lastIndexOf('/'));
+
+        const resolvedLibOutFolder: string = path.join(compiler.context, outFolderName);
+        const outputPathRelativeLibOutFolder: string = Path.convertToSlashes(
+          path.relative(jsFileFolderPath, resolvedLibOutFolder)
+        );
+        const libOutFolderRelativeOutputPath: string = Path.convertToSlashes(
+          path.relative(resolvedLibOutFolder, jsFilePath)
+        );
+        for (const libPath of libPaths) {
+          const depth: number = countSlashes(libPath);
+          const requirePath: string = '../'.repeat(depth) + libOutFolderRelativeOutputPath;
+          const moduleText: string = [
+            `module.exports = require('${requirePath}').getPath('${libPath}');`
+          ].join('\n');
+
+          compilation.emitAsset(
+            `${outputPathRelativeLibOutFolder}/${libPath}`,
+            new compiler.webpack.sources.RawSource(moduleText)
```

Comment:
If this is really expensive, would it be prudent to use [CachedSource](https://github.com/webpack/webpack-sources#cachedsource)?

Reply:
That actually feels like it'd probably add more overhead, honestly.